features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigBean.java

```diff
@@ -81,5 +85,26 @@ public boolean getNewSuspectOnTrap() {
 	public List<SnmpV3User> getSnmpV3Users() {
 		return m_snmpV3Users;
 	}
+	
+	public void onSnmpV3UsersUpdate(TrapdConfiguration config) {
+		this.m_snmpTrapAddress = config.getSnmpTrapAddress();
+		this.m_snmpTrapPort =config.getSnmpTrapPort();
+		this.m_snmpV3Users=addToSnmpV3User(config.getSnmpv3UserCollection());
```

Comment:
m_snmpV3Users needs to be protected with some kind of synchronization so that it is thread-safe. It's probably appropriate to use Collections.synchronizedList() and synchronized{} blocks to protect it since it won't be accessed frequently.